hv_go.hpp   mets::simulated_annealing<pcl::GlobalHypothesesVerification<pcl::PointXYZ, pcl::PointXYZ>::move_manager>' has no member named 'setApplyAndEvaluate'

[ 55%] Building CXX object recognition/CMakeFiles/pcl_recognition.dir/src/hv/hv_go.cpp.o
In file included from /home/acorbi/pcl/recognition/src/hv/hv_go.cpp:37:0:
/home/acorbi/pcl/recognition/include/pcl/recognition/impl/hv/hv_go.hpp: In instantiation of 'void pcl::GlobalHypothesesVerification<ModelT, SceneT>::SAOptimize(std::vector<int>&, std::vector<bool>&) [with ModelT = pcl::PointXYZ; SceneT = pcl::PointXYZ]':
/home/acorbi/pcl/recognition/src/hv/hv_go.cpp:42:1:   required from here
/home/acorbi/pcl/recognition/include/pcl/recognition/impl/hv/hv_go.hpp:451:3: error: 'class mets::simulated_annealing<pcl::GlobalHypothesesVerification<pcl::PointXYZ, pcl::PointXYZ>::move_manager>' has no member named 'setApplyAndEvaluate'
   sa.setApplyAndEvaluate(true);
   ^
/home/acorbi/pcl/recognition/include/pcl/recognition/impl/hv/hv_go.hpp: In instantiation of 'void pcl::GlobalHypothesesVerification<ModelT, SceneT>::SAOptimize(std::vector<int>&, std::vector<bool>&) [with ModelT = pcl::PointXYZRGB; SceneT = pcl::PointXYZRGB]':
/home/acorbi/pcl/recognition/src/hv/hv_go.cpp:43:1:   required from here
/home/acorbi/pcl/recognition/include/pcl/recognition/impl/hv/hv_go.hpp:451:3: error: 'class mets::simulated_annealing<pcl::GlobalHypothesesVerification<pcl::PointXYZRGB, pcl::PointXYZRGB>::move_manager>' has no member named 'setApplyAndEvaluate'
/home/acorbi/pcl/recognition/include/pcl/recognition/impl/hv/hv_go.hpp: In instantiation of 'void pcl::GlobalHypothesesVerification<ModelT, SceneT>::SAOptimize(std::vector<int>&, std::vector<bool>&) [with ModelT = pcl::PointXYZRGBA; SceneT = pcl::PointXYZRGBA]':
/home/acorbi/pcl/recognition/src/hv/hv_go.cpp:44:1:   required from here
/home/acorbi/pcl/recognition/include/pcl/recognition/impl/hv/hv_go.hpp:451:3: error: 'class mets::simulated_annealing<pcl::GlobalHypothesesVerification<pcl::PointXYZRGBA, pcl::PointXYZRGBA>::move_manager>' has no member named 'setApplyAndEvaluate'
make[2]: **\* [recognition/CMakeFiles/pcl_recognition.dir/src/hv/hv_go.cpp.o] Error 1
make[1]: **\* [recognition/CMakeFiles/pcl_recognition.dir/all] Error 2
make: **\* [all] Error 2

Quick note, this issue is still alive and well.  I'm attempting to build PCL on CentOS 6 and grabbed the latest version of metslib from source and built it (version 0.5.3) which results in:
`                 from /jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/recognition/src/hv/hv_go.cpp:37:
/jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/filters/include/pcl/filters/voxel_grid.h: In instantiation of ‘pcl::VoxelGrid<pcl::PointXYZRGBA>’:
/jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/recognition/include/pcl/recognition/impl/hv/hv_go.hpp:497:   instantiated from ‘bool pcl::GlobalHypothesesVerification<ModelT, SceneT>::addModel(typename pcl::PointCloud<PointT>::ConstPtr&, typename pcl::PointCloud<PointT>::ConstPtr&, boost::shared_ptr<pcl::GlobalHypothesesVerification<ModelT, SceneT>::RecognitionModel>&) [with ModelT = pcl::PointXYZRGBA, SceneT = pcl::PointXYZRGBA]’
/jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/recognition/src/hv/hv_go.cpp:44:   instantiated from here
/jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/filters/include/pcl/filters/voxel_grid.h:457: warning: ‘pcl::VoxelGrid<pcl::PointXYZRGBA>::leaf_size_’ contains empty classes which may cause base classes to be placed at different locations in a future version of GCC
/jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/filters/include/pcl/filters/voxel_grid.h:472: warning: ‘pcl::VoxelGrid<pcl::PointXYZRGBA>::min_b_’ contains empty classes which may cause base classes to be placed at different locations in a future version of GCC
/jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/filters/include/pcl/filters/voxel_grid.h:472: warning: ‘pcl::VoxelGrid<pcl::PointXYZRGBA>::div_b_’ contains empty classes which may cause base classes to be placed at different locations in a future version of GCC
In file included from /jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/recognition/src/hv/hv_go.cpp:37:
/jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/recognition/include/pcl/recognition/impl/hv/hv_go.hpp: In member function ‘void pcl::GlobalHypothesesVerification<ModelT, SceneT>::SAOptimize(std::vector<int, std::allocator<int> >&, std::vector<bool, std::allocator<bool> >&) [with ModelT = pcl::PointXYZRGBA, SceneT = pcl::PointXYZRGBA]’:
/jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/recognition/src/hv/hv_go.cpp:44:   instantiated from here
/jukebox/pkgs/PCL/src/pcl-pcl-1.7.2/recognition/include/pcl/recognition/impl/hv/hv_go.hpp:451: error: ‘class mets::simulated_annealing<pcl::GlobalHypothesesVerification<pcl::PointXYZRGBA, pcl::PointXYZRGBA>::move_manager>’ has no member named ‘setApplyAndEvaluate’
`

The cmake build doesn't note anything about a local version being available to build against just generates an error noting there isn't one available which is what prompted me to install 0.5.3.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -349,19 +349,6 @@ PCL_ADD_GRABBER_DEPENDENCY("DSSDK" "DepthSense SDK support")
 PCL_ADD_GRABBER_DEPENDENCY("RSSDK" "RealSense SDK support")
 PCL_ADD_GRABBER_DEPENDENCY("RSSDK2" "RealSense SDK 2.0 (librealsense) support")
 
-# metslib
-if(PKG_CONFIG_FOUND)
-  pkg_check_modules(METSLIB metslib)
-  if(METSLIB_FOUND)
-    set(HAVE_METSLIB ON)
-    include_directories(SYSTEM ${METSLIB_INCLUDE_DIRS})
-  else()
-    include_directories(SYSTEM "${PCL_SOURCE_DIR}/recognition/include/pcl/recognition/3rdparty/")
-  endif()
-else()
-    include_directories(SYSTEM ${PCL_SOURCE_DIR}/recognition/include/pcl/recognition/3rdparty/)
-endif()
-
 # LibPNG
 option(WITH_PNG "PNG file support" TRUE)
 if(WITH_PNG)

--- a/recognition/CMakeLists.txt
+++ b/recognition/CMakeLists.txt
@@ -141,21 +141,17 @@ set(srcs
   src/implicit_shape_model.cpp
 )
 
-if(HAVE_METSLIB)
-  set(metslib_incs "")
-else()
-  set(metslib_incs
-    "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/abstract-search.hh"
-    "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/local-search.hh"
-    "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/mets.hh"
-    "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/metslib_config.hh"
-    "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/model.hh"
-    "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/observer.hh"
-    "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/simulated-annealing.hh"
-    "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/tabu-search.hh"
-    "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/termination-criteria.hh"
-  )
-endif()
+set(metslib_incs
+  "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/abstract-search.hh"
+  "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/local-search.hh"
+  "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/mets.hh"
+  "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/metslib_config.hh"
+  "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/model.hh"
+  "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/observer.hh"
+  "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/simulated-annealing.hh"
+  "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/tabu-search.hh"
+  "include/pcl/${SUBSYS_NAME}/3rdparty/metslib/termination-criteria.hh"
+)
 
 set(LIB_NAME "pcl_${SUBSYS_NAME}")
 PCL_ADD_LIBRARY(${LIB_NAME} COMPONENT ${SUBSYS_NAME} SOURCES ${srcs} ${incs} ${impl_incs} ${face_detection_incs} ${ransac_based_incs} ${ransac_based_impl_incs} ${hv_incs} ${hv_impl_incs} ${cg_incs} ${cg_impl_incs} ${metslib_incs})
@@ -174,7 +170,4 @@ PCL_ADD_INCLUDES("${SUBSYS_NAME}" "${SUBSYS_NAME}/impl/hv" ${hv_impl_incs})
 PCL_ADD_INCLUDES("${SUBSYS_NAME}" "${SUBSYS_NAME}/impl/cg" ${cg_impl_incs})
 PCL_ADD_INCLUDES("${SUBSYS_NAME}" "${SUBSYS_NAME}/linemod" ${LINEMOD_INCLUDES})
 PCL_ADD_INCLUDES("${SUBSYS_NAME}" "${SUBSYS_NAME}/impl/linemod" ${LINEMOD_IMPLS})
-
-if(NOT HAVE_METSLIB)
-  PCL_ADD_INCLUDES("${SUBSYS_NAME}" "${SUBSYS_NAME}/3rdparty/metslib" ${metslib_incs})
-endif()
+PCL_ADD_INCLUDES("${SUBSYS_NAME}" "${SUBSYS_NAME}/3rdparty/metslib" ${metslib_incs})

